src/persistence/profile.cpp

```diff
@@ -105,9 +105,8 @@ Profile::Profile(QString name, const QString& password, bool isNewProfile,
     , encrypted{this->passkey != nullptr}
 {
     Settings& s = Settings::getInstance();
-    s.setCurrentProfile(name);
-    s.saveGlobal();
-    s.loadPersonal(name, this->passkey.get());
+    // TODO(kriby): Move/refactor core initialization to remove settings dependency
```

Comment:
My argument is as follows; In the long term, there is nothing about a user profile that implies it should hold elements of the tox connection itself. As a direct consequence of  Profile holding Core right now, the startup sequence code complexity goes up significantly. I imagine code complexity elsewhere in the project could be affected as well, at a minimum the tight coupling makes refactoring efforts to reduce singleton dependency in the project overall harder.

In the long term, I would argue that it would healthier for qTox as a project to separate class concerns concisely and ween off use of singleton access so that we can minimize unexpected adverse effects of code changes throughout the project. I added this todo in the spirit of that goal, but I admit it may take a while to reach a point where we can safely resolve it.

Reply:
> nothing about a user profile that implies it should hold elements of the tox connection itself

Yes, but we talk about settings.

I agree with all your thesis. In particular, that Settings singleton should be replaced on Dependency Injection. But in your comment, you said that settings dependency should be completely removed

Reply:
It was mostly meant as a reminder to myself to attempt to refactor that code to the effect I described at some point in the near future. I will remove it.